zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/BufferedProcessingResultBuilder.java

```diff
@@ -41,21 +41,33 @@ final class BufferedProcessingResultBuilder implements ProcessingResultBuilder {
   private final RecordBatch mutableRecordBatch;
   private ProcessingResponseImpl processingResponse;
   private final long operationReference;
+  private int currentSourceIndex;
 
   BufferedProcessingResultBuilder(final RecordBatchSizePredicate predicate) {
     this(predicate, operationReferenceNullValue());
   }
 
   BufferedProcessingResultBuilder(
       final RecordBatchSizePredicate predicate, final long operationReference) {
+    this(predicate, operationReferenceNullValue(), -1);
+  }
+
+  BufferedProcessingResultBuilder(
+      final RecordBatchSizePredicate predicate,
+      final long operationReference,
+      final int currentSourceIndex) {
     mutableRecordBatch = new RecordBatch(predicate);
     this.operationReference = operationReference;
+    this.currentSourceIndex = currentSourceIndex;
+  }
+
+  public void setCurrentSourceIndex(final int currentSourceIndex) {
```

Comment:
:wrench: This could be package-private, no? Because we don't want to give everyone the ability to do this, pretty much only the ProcessingStateMachine is allowed.

Reply:
Agreed. Removed the public modifier.